// react-native-google-mobile-ads/lib/commonjs/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
var _exportNames = {
  SDK_VERSION: true,
  MobileAds: true,
  AdsConsentDebugGeography: true,
  AdsConsentInfo: true,
  AdsConsentInfoOptions: true,
  AdsConsentInterface: true,
  AdsConsentPrivacyOptionsRequirementStatus: true,
  AdsConsentStatus: true,
  AdsConsentUserChoices: true,
  AdsConsentPurposes: true,
  AdsConsentSpecialFeatures: true,
  MaxAdContentRating: true,
  TestIds: true,
  AdEventType: true,
  BannerAdSize: true,
  GAMBannerAdSize: true,
  GAMAdEventType: true,
  NativeAdEventType: true,
  RewardedAdEventType: true,
  AdsConsent: true,
  AppOpenAd: true,
  InterstitialAd: true,
  RewardedAd: true,
  RewardedInterstitialAd: true,
  BannerAd: true,
  NativeAd: true,
  NativeAdView: true,
  NativeMediaView: true,
  NativeAsset: true,
  NativeAssetType: true,
  GAMBannerAd: true,
  GAMInterstitialAd: true,
  useAppOpenAd: true,
  useInterstitialAd: true,
  useRewardedAd: true,
  useRewardedInterstitialAd: true,
  useForeground: true,
  RevenuePrecisions: true
};
Object.defineProperty(exports, "AdEventType", {
  enumerable: true,
  get: function () {
    return _AdEventType.AdEventType;
  }
});
Object.defineProperty(exports, "AdsConsent", {
  enumerable: true,
  get: function () {
    return _AdsConsent.AdsConsent;
  }
});
Object.defineProperty(exports, "AdsConsentDebugGeography", {
  enumerable: true,
  get: function () {
    return _NativeConsentModule.AdsConsentDebugGeography;
  }
});
Object.defineProperty(exports, "AdsConsentInfo", {
  enumerable: true,
  get: function () {
    return _NativeConsentModule.AdsConsentInfo;
  }
});
Object.defineProperty(exports, "AdsConsentInfoOptions", {
  enumerable: true,
  get: function () {
    return _NativeConsentModule.AdsConsentInfoOptions;
  }
});
Object.defineProperty(exports, "AdsConsentInterface", {
  enumerable: true,
  get: function () {
    return _NativeConsentModule.AdsConsentInterface;
  }
});
Object.defineProperty(exports, "AdsConsentPrivacyOptionsRequirementStatus", {
  enumerable: true,
  get: function () {
    return _NativeConsentModule.AdsConsentPrivacyOptionsRequirementStatus;
  }
});
Object.defineProperty(exports, "AdsConsentPurposes", {
  enumerable: true,
  get: function () {
    return _AdsConsentPurposes.AdsConsentPurposes;
  }
});
Object.defineProperty(exports, "AdsConsentSpecialFeatures", {
  enumerable: true,
  get: function () {
    return _AdsConsentSpecialFeatures.AdsConsentSpecialFeatures;
  }
});
Object.defineProperty(exports, "AdsConsentStatus", {
  enumerable: true,
  get: function () {
    return _NativeConsentModule.AdsConsentStatus;
  }
});
Object.defineProperty(exports, "AdsConsentUserChoices", {
  enumerable: true,
  get: function () {
    return _NativeConsentModule.AdsConsentUserChoices;
  }
});
Object.defineProperty(exports, "AppOpenAd", {
  enumerable: true,
  get: function () {
    return _AppOpenAd.AppOpenAd;
  }
});
Object.defineProperty(exports, "BannerAd", {
  enumerable: true,
  get: function () {
    return _BannerAd.BannerAd;
  }
});
Object.defineProperty(exports, "BannerAdSize", {
  enumerable: true,
  get: function () {
    return _BannerAdSize.BannerAdSize;
  }
});
Object.defineProperty(exports, "GAMAdEventType", {
  enumerable: true,
  get: function () {
    return _GAMAdEventType.GAMAdEventType;
  }
});
Object.defineProperty(exports, "GAMBannerAd", {
  enumerable: true,
  get: function () {
    return _GAMBannerAd.GAMBannerAd;
  }
});
Object.defineProperty(exports, "GAMBannerAdSize", {
  enumerable: true,
  get: function () {
    return _BannerAdSize.GAMBannerAdSize;
  }
});
Object.defineProperty(exports, "GAMInterstitialAd", {
  enumerable: true,
  get: function () {
    return _GAMInterstitialAd.GAMInterstitialAd;
  }
});
Object.defineProperty(exports, "InterstitialAd", {
  enumerable: true,
  get: function () {
    return _InterstitialAd.InterstitialAd;
  }
});
Object.defineProperty(exports, "MaxAdContentRating", {
  enumerable: true,
  get: function () {
    return _MaxAdContentRating.MaxAdContentRating;
  }
});
Object.defineProperty(exports, "MobileAds", {
  enumerable: true,
  get: function () {
    return _MobileAds.MobileAds;
  }
});
Object.defineProperty(exports, "NativeAd", {
  enumerable: true,
  get: function () {
    return _NativeAd.NativeAd;
  }
});
Object.defineProperty(exports, "NativeAdEventType", {
  enumerable: true,
  get: function () {
    return _NativeAdEventType.NativeAdEventType;
  }
});
Object.defineProperty(exports, "NativeAdView", {
  enumerable: true,
  get: function () {
    return _NativeAdView.NativeAdView;
  }
});
Object.defineProperty(exports, "NativeAsset", {
  enumerable: true,
  get: function () {
    return _NativeAsset.NativeAsset;
  }
});
Object.defineProperty(exports, "NativeAssetType", {
  enumerable: true,
  get: function () {
    return _NativeAsset.NativeAssetType;
  }
});
Object.defineProperty(exports, "NativeMediaView", {
  enumerable: true,
  get: function () {
    return _NativeMediaView.NativeMediaView;
  }
});
Object.defineProperty(exports, "RevenuePrecisions", {
  enumerable: true,
  get: function () {
    return _constants.RevenuePrecisions;
  }
});
Object.defineProperty(exports, "RewardedAd", {
  enumerable: true,
  get: function () {
    return _RewardedAd.RewardedAd;
  }
});
Object.defineProperty(exports, "RewardedAdEventType", {
  enumerable: true,
  get: function () {
    return _RewardedAdEventType.RewardedAdEventType;
  }
});
Object.defineProperty(exports, "RewardedInterstitialAd", {
  enumerable: true,
  get: function () {
    return _RewardedInterstitialAd.RewardedInterstitialAd;
  }
});
exports.SDK_VERSION = void 0;
Object.defineProperty(exports, "TestIds", {
  enumerable: true,
  get: function () {
    return _TestIds.TestIds;
  }
});
Object.defineProperty(exports, "default", {
  enumerable: true,
  get: function () {
    return _MobileAds.default;
  }
});
Object.defineProperty(exports, "useAppOpenAd", {
  enumerable: true,
  get: function () {
    return _useAppOpenAd.useAppOpenAd;
  }
});
Object.defineProperty(exports, "useForeground", {
  enumerable: true,
  get: function () {
    return _useForeground.useForeground;
  }
});
Object.defineProperty(exports, "useInterstitialAd", {
  enumerable: true,
  get: function () {
    return _useInterstitialAd.useInterstitialAd;
  }
});
Object.defineProperty(exports, "useRewardedAd", {
  enumerable: true,
  get: function () {
    return _useRewardedAd.useRewardedAd;
  }
});
Object.defineProperty(exports, "useRewardedInterstitialAd", {
  enumerable: true,
  get: function () {
    return _useRewardedInterstitialAd.useRewardedInterstitialAd;
  }
});
var _version = require("./version");
var _MobileAds = _interopRequireWildcard(require("./MobileAds"));
var _NativeConsentModule = require("./specs/modules/NativeConsentModule");
var _AdsConsentPurposes = require("./AdsConsentPurposes");
var _AdsConsentSpecialFeatures = require("./AdsConsentSpecialFeatures");
var _MaxAdContentRating = require("./MaxAdContentRating");
var _TestIds = require("./TestIds");
var _AdEventType = require("./AdEventType");
var _BannerAdSize = require("./BannerAdSize");
var _GAMAdEventType = require("./GAMAdEventType");
var _NativeAdEventType = require("./NativeAdEventType");
var _RewardedAdEventType = require("./RewardedAdEventType");
var _AdsConsent = require("./AdsConsent");
var _AppOpenAd = require("./ads/AppOpenAd");
var _InterstitialAd = require("./ads/InterstitialAd");
var _RewardedAd = require("./ads/RewardedAd");
var _RewardedInterstitialAd = require("./ads/RewardedInterstitialAd");
var _BannerAd = require("./ads/BannerAd");
var _NativeAd = require("./ads/native-ad/NativeAd");
var _NativeAdView = require("./ads/native-ad/NativeAdView");
var _NativeMediaView = require("./ads/native-ad/NativeMediaView");
var _NativeAsset = require("./ads/native-ad/NativeAsset");
var _GAMBannerAd = require("./ads/GAMBannerAd");
var _GAMInterstitialAd = require("./ads/GAMInterstitialAd");
var _useAppOpenAd = require("./hooks/useAppOpenAd");
var _useInterstitialAd = require("./hooks/useInterstitialAd");
var _useRewardedAd = require("./hooks/useRewardedAd");
var _useRewardedInterstitialAd = require("./hooks/useRewardedInterstitialAd");
var _useForeground = require("./hooks/useForeground");
var _constants = require("./common/constants");
var _types = require("./types");
Object.keys(_types).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _types[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _types[key];
    }
  });
});
function _interopRequireWildcard(e, t) { if ("function" == typeof WeakMap) var r = new WeakMap(), n = new WeakMap(); return (_interopRequireWildcard = function (e, t) { if (!t && e && e.__esModule) return e; var o, i, f = { __proto__: null, default: e }; if (null === e || "object" != typeof e && "function" != typeof e) return f; if (o = t ? n : r) { if (o.has(e)) return o.get(e); o.set(e, f); } for (const t in e) "default" !== t && {}.hasOwnProperty.call(e, t) && ((i = (o = Object.defineProperty) && Object.getOwnPropertyDescriptor(e, t)) && (i.get || i.set) ? o(f, t, i) : f[t] = e[t]); return f; })(e, t); }
/*
 * Copyright (c) 2016-present Invertase Limited & Contributors
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this library except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 */

// import { SDK_VERSION } from 'react-native-google-mobile-ads';
const SDK_VERSION = exports.SDK_VERSION = _version.version;
