// react-native-google-mobile-ads/lib/commonjs/NativeAdEventType.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.NativeAdEventType = void 0;
let NativeAdEventType = exports.NativeAdEventType = /*#__PURE__*/function (NativeAdEventType) {
  /**
   * Called when an impression is recorded for an ad.
   */
  NativeAdEventType["IMPRESSION"] = "impression";
  /**
   * Called when a click is recorded for an ad.
   */
  NativeAdEventType["CLICKED"] = "clicked";
  /**
   * Called when an ad opens an overlay that covers the screen.
   */
  NativeAdEventType["OPENED"] = "opened";
  /**
   * Called when the user is about to return to the application after clicking on an ad.
   */
  NativeAdEventType["CLOSED"] = "closed";
  /**
   * Called when an ad is estimated to have earned money.
   */
  NativeAdEventType["PAID"] = "paid";
  /**
   * Called when the video controller has begun or resumed playing a video
   */
  NativeAdEventType["VIDEO_PLAYED"] = "video_played";
  /**
   * Called when the video controller has paused video.
   */
  NativeAdEventType["VIDEO_PAUSED"] = "video_paused";
  /**
   * Called when the video controller’s video playback has ended.
   */
  NativeAdEventType["VIDEO_ENDED"] = "video_ended";
  /**
   * Called when the video controller has muted video.
   */
  NativeAdEventType["VIDEO_MUTED"] = "video_muted";
  /**
   * Called when the video controller has unmuted video.
   */
  NativeAdEventType["VIDEO_UNMUTED"] = "video_unmuted";
  return NativeAdEventType;
}({});
